src/components/VContentReport/VContentReportForm.vue

```diff
@@ -142,20 +151,30 @@ export default defineComponent({
     VReportDescForm,
   },
   props: {
-    media: { required: true },
-    providerName: { required: true },
-    reportService: { required: false },
-    closeFn: { required: true },
+    media: {
+      type: Object as PropType<Media>,
+      required: true,
+    },
+    providerName: {
+      type: String,
+      required: true,
+    },
+    reportService: {
+      type: Object as PropType<{ sendReport: () => Promise<void> }>,
+      required: false,
+    },
+    closeFn: {
+      type: Object as PropType<() => void>,
+      required: true,
+    },
   },
   setup(props) {
     const service = props.reportService || ReportService
     const description = ref('')
 
-    /** @type {import('@nuxtjs/composition-api').Ref<string|null>} */
-    const status = ref(WIP)
+    const status = ref<ReportStatus | null>(WIP)
 
-    /** @type {import('@nuxtjs/composition-api').Ref<string|null>} */
-    const selectedReason = ref(DMCA)
+    const selectedReason = ref<ReportReason | null>(DMCA)
```

Comment:
This is the other thing, you will get totally bogus errors because VSCode is running an improperly/incompletely configured TypeScript :confused: 

I think the rest of the changes in this PR are good but adding `lang="ts"` and converting them to TS I think should wait until the files can actually be added to the tsconfig.